Mathlib/CategoryTheory/Sites/Sieves.lean

```diff
@@ -71,6 +71,30 @@ abbrev cocone (S : Presieve X) : Cocone S.diagram :=
 def bind (S : Presieve X) (R : ∀ ⦃Y⦄ ⦃f : Y ⟶ X⦄, S f → Presieve Y) : Presieve X := fun Z h =>
   ∃ (Y : C) (g : Z ⟶ Y) (f : Y ⟶ X) (H : S f), R H g ∧ g ≫ f = h
 
+/-- Structure which contains the data and properties for a morphism `h` satisfying
+`Presieve.bind S R h`. -/
+structure BindStruct (S : Presieve X) (R : ∀ ⦃Y⦄ ⦃f : Y ⟶ X⦄, S f → Presieve Y)
+    {Z : C} (h : Z ⟶ X) where
+  /-- the intermediate object -/
+  Y : C
+  /-- a morphism in the family of presieves `R` -/
+  g : Z ⟶ Y
+  /-- a morphism in the presieve `S` -/
+  f : Y ⟶ X
+  hf : S f
+  hg : R hf g
+  fac : g ≫ f = h
+
+attribute [reassoc (attr := simp)] BindStruct.fac
+
+/-- If a morphism `h` satisfies `Presieve.bind S R h`, this is a choice of a structure
+in `BindStruct S R h`. -/
+noncomputable def bind.bindStruct {S : Presieve X} {R : ∀ ⦃Y⦄ ⦃f : Y ⟶ X⦄, S f → Presieve Y}
+    {Z : C} {h : Z ⟶ X} (H : bind S R h) : BindStruct S R h :=
+  Nonempty.some (by
```

Comment:
Maybe also the converse is useful? I.e.
```
lemma BindStruct.bind : {S : Presieve X} {R : ∀ ⦃Y⦄ ⦃f : Y ⟶ X⦄, S f → Presieve Y}
    {Z : C} {h : Z ⟶ X} (b : BindStruct S R h) : bind S R h :=
  ⟨b.Y, b.g, b.f, b.hf, b.hg, b.fac⟩
```
(not tested)

Reply:
Thanks for the suggestion! I have just add it.